docs/src/main/sphinx/security/file-system-access-control.rst

```diff
@@ -331,6 +347,93 @@ The example below defines the following table access policy:
       ]
     }
 
+.. _system-file-function-rules:
+
+Function rules
+^^^^^^^^^^^^^^
+
+These rules control the ability of a user to execute SQL functions, including
+PTFs (Polymorphic Table Functions).
+
+Each function rule is composed of the following fields:
+
+* ``user`` (optional): regex to match against user name. Defaults to ``.*``.
+* ``role`` (optional): regex to match against role names. Defaults to ``.*``.
+* ``group`` (optional): regex to match against group names. Defaults to ``.*``.
+* ``catalog`` (optional): regex to match against catalog name. Defaults to
+  ``.*``.
+* ``schema`` (optional): regex to match against schema name. Defaults to ``.*``.
+* ``function`` (optional): regex to match against function names. Defaults to
+  ``.*``.
+* ``privileges`` (required): zero or more of ``EXECUTE``, ``GRANT_EXECUTE``.
+* ``function_kinds`` (required): one or more of ``AGGREGATE``, ``SCALAR``,
+  ``TABLES``, ``WINDOW``. When user define a rule for ``AGGREGATE``, ``SCALAR``
+  or ``WINDOW`` function, ``catalog`` and ``schema`` fields are disallowed.
+
+For example, to restore previous behavior (all Polymorphic Table Functions from any
+catalogs is denied) you can use the following rules:
+
+.. code-block:: json
+
+    {
+      "functions": [
+        {
+          "privileges": [
+            "EXECUTE",
+            "GRANT_EXECUTE"
+          ],
+          "function_kinds": [
+            "SCALAR",
+            "AGGREGATE",
+            "WINDOW"
+          ]
+        },
+        {
+          "privileges": [],
+          "function_kinds": [
+            "TABLE"
+          ]
+        }
+      ]
+    }
+
+To allow executing ``query`` polymorphic table function from any catalogs by
+``admin`` user.
```

Comment:
```suggestion
To allow use of the ``query`` polymorphic table function from any catalog by
an ``admin`` user:
```

Reply:
I used the suggestion from https://github.com/trinodb/trino/pull/14774#discussion_r1009706978